server/server.cc

```diff
@@ -41,15 +45,51 @@ void Session::do_read() {
     [this, self](boost::system::error_code ec, std::size_t len) {
       if (!ec) {
         printf("Incoming Data length %lu:\n", len);
-        do_write();
+        std::unique_ptr<Request> req = Request::Parse(data_);
+        Response* response = new Response;
+        std::string response_string;
+        if(req.get()) {
+          // response valid 
+          printf("request key: %s\n", req->uri().c_str());
+          std::string key = "";
+          for(const auto & key_match : *handler_->RequestHandlers) {
+            auto res = std::mismatch(key_match.first.begin(), key_match.first.end(), req->uri().begin());
+            if(res.first == key_match.first.end()) { //match found
+              std::string tmp_key = std::string(key_match.first.begin(), res.first);
+              if(tmp_key.size() > key.size()) { //if the match is better, that's our new key'
+                
+                key = tmp_key;
```

Comment:
I'll call it best_key.

I don't know why you would have to go between files to understand what key_match is?